javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithStatements.java

```diff
@@ -71,8 +70,20 @@ default ExpressionStmt addAndGetStatement(Expression expr) {
     default ExpressionStmt addAndGetStatement(String statement) {
         return addAndGetStatement(new NameExpr(statement));
     }
-    
+
     default boolean isEmpty() {
         return getStmts().isEmpty();
     }
+
+    @SuppressWarnings("unchecked")
+    default N copyStatements(NodeList<Statement> nodeList) {
+        for (Statement n : nodeList) {
+            addStatement((Statement) n.clone());

```

Comment:
It actually make a shallow copy, do we have somthing to deep copy statements ? (I tried using CloneVisitor but without success)